packages/vm/src/context.rs

```diff
@@ -155,7 +155,7 @@ where
     }
 }
 
-pub(crate) fn with_querier_from_context<'a, 'b, S, Q, F, T>(
+pub(crate) fn with_querier_from_context<'a, 'b, S, Q: 'b, F, T>(
```

Comment:
Huh, `Q: 'b` syntax... this means that `Q` has at least lifetime `'b`?

Reply:
Yes, the `x: y` means _x lives at least as long as y_. This works for lifetimes `'b: 'a` or generic types like above.

This change became necessary and was suggested by the compiler when removing `'static` from `pub trait Querier:`

------------

To be honest, I don't understand what `pub trait Querier: 'static {` means. We used it before, but was it on purpose or to silence the compiler? From how I understand the docs it means: every `Querier` outlives static lifetime. @reuvenpo do you know that?

Reply:
being constrained by `'static` means that the type has no lifetmies associated with it. it does not necessarily mean no internal pointers - for example a `Box<T>`, or an object holding it, is also `'static`.
When writing generic code where e.g. `T: SomeTrait`, the compiler assumes that `T` may contain a lifetime (if it contains references) and forces generic code to handle that property. To constrain the amount of checks the compiler performs, you can tell it that either the trait requires that implementors are `'static` (`trait SomeTrait: 'static { ... }`, or that the specific object is static, at function declaration (`T: SomeTrait + 'static`).

Similarly, if you need to refer to the lifetime that the compiler permits for the generic type, you can name it like so: `<'a, T: SomeTrait + 'a, ... >`

I hope this made sense

Reply:
I think I get it mostly, thanks!

> you can tell it that either the trait requires that implementors are `'static` (`trait SomeTrait: 'static { ... }`

Just to double check: Using `pub trait Querier: 'static` adds additional lifetime restrictions to the implementations of `Querier` that `pub trait Querier` does not have?

Reply:
yes, it requires that all implementors do not hold references to objects they do not own, effectively